### DISCUSSION TITLE
Is it possible to define absorbing boundaries for a saturated material?

### DISCUSSION THREAD

**Discussion by albuquerqueraniel** [Anura3D/Anura3D_OpenSource#82] [Q&A] [answered]
Hello, everyone!

I'm trying to define absorbing boundaries in a saturated material and I have a few questions about it. The model is a variation of the Tutorial Manual's Impact Problem, as you can see in the picture:

![0](https://user-images.githubusercontent.com/45863863/225996786-ceaa5f86-1de1-4160-9183-8e35a13511ee.png)

Should I define an absorbing boundary on the right edge of the moving mesh (circled in the next picture)? How, as the body penetrates the soil, it is going to overlay the bottom right edge I was wondering if this edge has to be defined as an absorbing boundary too.

![1](https://user-images.githubusercontent.com/45863863/225995947-30179466-ef49-4ae9-9957-8715844e781d.png) 

Unlike the fixities window that has Solid **and** Liquid Fixities, the absorbing boundaries window has just Solid **or** Liquid boundaries. I've defined the solid and liquid boundaries individually (one after another) for the same instances. GiD didn't show me an error when I generated Anura3D files.

When Anura calculated the results, I was receiving the following error: 

![4](https://user-images.githubusercontent.com/45863863/226001636-25969ce9-3742-4db3-ba20-524cd63a82eb.png)

I thought that maybe this particle is leaving the mesh because of a wrong definition of the absorbing boundaries, so I went to the GOM file to check how the boundaries were defined there. But the GOM file has just the properties of the solid absorbing boundaries.

It is a bug while generating the GOM file or I really can't define absorbing boundaries for both phases?

If I can't define absorbing boundaries for both phases, it is a theoretical limitation or it was just not implemented yet?

Best regards, 
Raniel

**Comment by FrancescaCeccato** [comment, marked as answer]
Hi,
Absorbing boundary is implemented for all phases. it can be assigned separately for solid and liquid as you did and this also works for all boundary conditions.
Also the boundary in the green circle should be an absorbing boudnary or a fixity (depending on what you want to model). 
Regarding the issue with partile leaving the mesh, this may be due to several causes, for example:

1. Absorbing boundary is not enough stiff, 
2. boundary conditions are not applied correclty everywhere needed
3. boundary is not far enough

I hope this could help.

Francesca

**Reply by albuquerqueraniel** [reply]
Thank you, Francesca.

I've reduced the alpha value (from 0.8 to 0.4) then the particle stopped leaving the domain. But I didn't understand why. I've searched for more details about these two parameters in the Scientific Manual and Al-Kafaji thesis.

The Scientific Manual and Al-Kafaji thesis have information, explicitly, just about the Delta, which is the thickness of the virtual layer, and it is used to adjust the spring coefficient. But in both references, it doesn't describes Alpha. It is the "a" in the dashpot coefficient of the Kelvin-Voigt element?

![kelvin-Voigt element](https://user-images.githubusercontent.com/45863863/231524810-e8dcf071-2282-48b8-aafd-f375c759c26e.png)


![alpha](https://user-images.githubusercontent.com/45863863/229591357-33296f76-33a4-4a7f-abce-4b3ab8d2ab09.png)


Best regards!
Raniel

**Reply by FrancescaCeccato** [reply]
Yes, delta adjust the spring stiffness and alpha adjust the viscous element.

**Reply by PouyaCR** [reply]
@albuquerqueraniel Hi, may I ask you how can we reduce the alpha value??

Regards

**Comment by FrancescaCeccato** [comment]
Alpha and delta are parameters that are assigned via GiD as input when assignin the absorbing boundary.

**Reply by PouyaCR** [reply]
Dear Ceccato
When I define absorbing boundary, I face with this error during analyses. Do you know how can I fix it?
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/145875446/06eddbf5-bdbc-4348-8a39-6ca2f3a036af)

**Reply by FrancescaCeccato** [reply]
this type of error is related to wrong reading of GOM file, no necessarily related to AB. Maybe something is wrong is the definition of contact or other conditions? usually by remaking the model from beginning, by opening a completely new GiD project is can be solved.

**Reply by CPLA81192** [reply]
What factors determine the values of alpha and Delta, and how can they be determined

**Comment by PouyaCR** [comment]
Hi
Whenever I use an absorbing boundary, I face an error during calculation.
Actually it does not start.

Regards
Pouya Zahedi
PhD Student
Department of Civil, Structural and Environmental Engineering
Trinity College Dublin, The University of Dublin
Dublin
Ireland
Linkedin <https://www.linkedin.com/in/pouya-zahedi-489651242>/ICRAG
<https://www.icrag-centre.org/people/pouyazahedi.html>

**Comment by PouyaCR** [comment]
But, whenever I unassign the AB, it starts to analyze the model.

**Reply by FrancescaCeccato** [reply]
Ok, then it will be very easy to debug the code. You can identify teh line where it crashes and we can fix the eventual problem.

**Comment by PouyaCR** [comment]
I have identified the line here in this pic
[image: image.png]
Pouya Zahedi
PhD Student
Department of Civil, Structural and Environmental Engineering
Trinity College Dublin, The University of Dublin
Dublin
Ireland
Linkedin <https://www.linkedin.com/in/pouya-zahedi-489651242>/ICRAG
<https://www.icrag-centre.org/people/pouyazahedi.html>
